src/components/modal/Modal.tsx

```diff
@@ -128,7 +133,7 @@ const Modal = ({
   return (
     <>
       {showContent && (
-        <div className="m-0">
+        <div className="m-0" onMouseDown={(e) => stopMouseDownPropagation && e.stopPropagation()} role='modal'>
```

Comment:
Consider extracting o avoid create new function on every render. Also change role='modal' to role="dialog" aria-modal="true" for ARIA compliance